src/controller.ts

```diff
@@ -71,6 +79,116 @@ export default function (
       }
     });
 
+    fastify.post<{ Body: { name: string; email: string } }>(
+      '/create-customer',
+      {
+        schema: {
+          body: {
+            type: 'object',
+            required: ['email', 'name'],
+            properties: { name: { type: 'string' }, email: { type: 'string' } },
+          },
+        },
+      },
+      async (req, res) => {
+        const { name, email } = req.body;
+
+        if (!email) {
+          return res.status(404).send({
+            message: 'Email should be provided',
+          });
+        }
+
+        try {
+          const createdCustomer = await paymentService.createCustomer({
+            name,
+            email,
+          });
+
+          return res.send({
+            customerId: createdCustomer.id,
+          });
+        } catch (err) {
+          return res.status(500).send({
+            message: 'Internal Server Error',
+          });
+        }
+      },
+    );
+
+    fastify.get<{
+      Querystring: { email: string };
+      schema: {
+        querystring: {
+          type: 'object';
+          properties: { email: { type: 'number' } };
+        };
+      };
+    }>('/get-customer-id', async (req, res) => {
+      const { email } = req.query;
+
+      if (!email) {
+        return res.status(404).send({
+          message: 'Email should be provided',
+        });
+      }
+
+      try {
+        const { id } = await paymentService.getCustomerIdByEmail(email);
```

Comment:
Validate that the user is not found on `getCustomerIdByEmail`. If the error is not a custom one or expected to happen, send a 500 and log the error